src/gc/gcinterface.h

```diff
@@ -691,6 +691,14 @@ class IGCHeap {
     // Indicates that an object's finalizer should be run upon the object's collection.
     virtual bool RegisterForFinalization(int gen, Object* obj) = 0;
 
+    virtual int GetLastGCTimeInGC() = 0;
+
+    virtual uint64_t GetLastGCGenerationSize(int gen) = 0;
+
+    virtual uint64_t GetTotalAllocation() = 0;
```

Comment:
>GetTotalAllocation [](start = 21, length = 18)

GetTotalAllocation sounds like you are getting all allocations so far. should call them something like GetLastGCAllocated, GetLastGCAllocatedLOH.

Reply:
That is what I am getting though.. Should I change it to return what was allocated since last GC?

Reply:
my bad! I read it wrong - the original perf counter value was indeed an accumulative value instead of for each GC (it's been a long time since I thought about perf counters at all :)). 

we do have another API already on the IGCHeap interface called GetTotalAllocatedBytes which gets the total and it does have the soh/loh separation. this however works a little differently as in it isn't calculated at the beginning of a GC but if you call it at the beginning of a GC it should return the same thing. if you want the separation you could add another API that just gets the LOH allocated bytes.

Reply:
Ah, I see. I didn't know we had that in IGCHeap - Thanks for the pointer! 

As far as I know I don't think it makes a big difference whether the stats reported was updated at the beginning of the GC / end of the GC. I'll just add another API for getting the LOH allocated bytes and remove the ones I added then.